File: api/my_function.py
# api/my_function.py

from flask import Flask, request

app = Flask(__name__)

@app.route('/api/my_function', methods=['GET'])
def my_function():
    name = request.args.get('name', 'World')
    message = f'Hello, {name}!'
    return message, 200

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=4000)

File: pages/api/my_function.js
// pages/api/my_function.js
import axios from 'axios';

export default async function handler(req, res) {
  const { name } = req.query;
  const port = process.env.NODE_ENV === 'production' ? 3000 : 4000;
  try {
    const response = await axios.get(`http://127.0.0.1:${port}/api/my_function`, { params: { name: name || 'World' } });
    res.status(response.status).send(response.data);
  } catch (error) {
    res.status(error.response?.status || 500).send(error.message);
  }
}
